atd-vzd/schema/afd__incidents.sql

```diff
@@ -0,0 +1,65 @@
+DROP TABLE public.afd__incidents;
+
+-- Table Definition
+CREATE TABLE "public"."afd__incidents" (
+    id serial primary key,
+    incident_number integer,
+    crash_id integer default null,
+    unparsed_ems_incident_number text default null,
+    ems_incident_numbers integer[] default null,
+    call_datetime timestamp default null,
+    calendar_year text default null,
+    jurisdiction text default null,
+    address text default null,
+    problem text default null,
+    flagged_incs text default null,
+    geometry geometry(Point,4326) default null,
+    austin_full_purpose bool default null,
+    location_id text default null,
+    latitude float8 default null,
+    longitude float8 default null,
+    ems_incident_number_1 integer default null,
+    ems_incident_number_2 integer default null,
+    call_date date default null,
+    call_time time default null 
+);
+
+CREATE OR REPLACE FUNCTION afd_incidents_trigger()
+  RETURNS trigger AS
+$$
+BEGIN
+  update afd__incidents set
+    austin_full_purpose = (
+      select ST_Contains(jurisdiction.geometry, incidents.geometry)
+      from afd__incidents incidents
+      left join atd_jurisdictions jurisdiction on (jurisdiction.jurisdiction_label = 'AUSTIN FULL PURPOSE')
+      where incidents.id = new.id),
+    location_id = (
+      select locations.location_id
+      from afd__incidents incidents
+      join atd_txdot_locations locations on (incidents.geometry && locations.shape and ST_Contains(locations.shape, incidents.geometry))
+      where incidents.id = new.id),
+    latitude = ST_Y(afd__incidents.geometry),
+    longitude = ST_X(afd__incidents.geometry),
+    ems_incident_number_1 = afd__incidents.ems_incident_numbers[1],
+    ems_incident_number_2 = afd__incidents.ems_incident_numbers[2],
+    call_date = date(afd__incidents.call_datetime),
+    call_time = afd__incidents.call_datetime::time
+    where afd__incidents.id = new.id;
+RETURN NEW;
+END;
+$$
+LANGUAGE 'plpgsql';
+
+create trigger afd_incidents_trigger_insert
+after insert on public.afd__incidents
+for each row execute procedure afd_incidents_trigger();
+
+create trigger afd_incidents_trigger_update
+after update on public.afd__incidents
+for each row 
+WHEN (false 
+  or OLD.geometry IS DISTINCT FROM NEW.geometry
+  or old.ems_incident_numbers is distinct from new.ems_incident_numbers
+  or old.call_datetime is distinct from new.call_datetime)
```

Comment:
This causes the trigger to only update on real changes to a certain set of columns. Same in the other SQL file.

Reply:
Cool - thanks for the explanation here. I haven't worked with PG triggers too much so it is nice to see how to manage them likes this.

Reply:
same -- thanks for the explanation. Why is it WHEN (false or x or y or z) and not just WHEN (x or y or z)? Convention?

Reply:
@chiaberry wrote: 
> Why is it WHEN (false or x or y or z) and not just WHEN (x or y or z)? Convention?

This is really just a Frank thing, but I like each of the three conditions that make up the larger `where` clause to line-up and have the same keywords around them. I think this better indicates that they are part of a working-group, and you can also comment them in and out without having to worry about if you have the right keywords.

I don't see this used by many folks except myself, so if you all think it's not a great pattern, I would be happy to change it.